test/users.js

```
@@ -0,0 +1,141 @@
+const chai = require('chai'),
+  server = require('./../app'),
+  userService = require('./../app/services/users'),
+  bcrypt = require('bcrypt'),
+  should = chai.should();
+
+describe('/users POST', () => {
+  it('should fail because email have a invalid host mail', (done) => {
+    return chai.request(server)
+    .post('/users')
+    .send({
+      firstName:'Jonh',
+      lastName:'Salchichon',
+      email:'JS@gmail.com',
+      password:'abcde12345'
+    }).catch((err) => {
+      err.should.have.status(400);
+      err.response.should.be.json;
+      err.response.body.error.should.equal('invalid email');
+    }).then(() => done());
+  });
+
+  it('should fail because password a invalid format', (done) => {
+    return chai.request(server)
+    .post('/users')
+    .send({
+      firstName:'Jonh',
+      lastName:'Salchichon',
+      email:'JS@wolox.com.ar',
+      password:'abcde@12345'
+    }).catch((err) => {
+      err.should.have.status(400);
+      err.response.should.be.json;
+      err.response.body.error.should.equal('invalid password');
+    }).then(() => done());
+  });
+
+  it('should fail because password is short ', (done) => {
+    return chai.request(server)
+    .post('/users')
+    .send({
+      firstName:'Jonh',
+      lastName:'Salchichon',
+      email:'JS@wolox.com.ar',
+      password:'1234'
+    }).catch((err) => {
+      err.should.have.status(400);
+      err.response.should.be.json;
+      err.response.body.error.should.equal('invalid password');
+    }).then(() => done());
+  });
+
+  it('should fail bacause the app don\'t recibe firstName', (done) => {
```

'should fail bacause the app doesn't recibe firstName' or 'should fail bacause firstName is missing'